migrations/2025-07-10-105316_username-instance-unique/up.sql

```diff
@@ -0,0 +1,10 @@
+-- lemmy requires (username + instance_id) to be unique
+-- delete any existing duplicates
+DELETE FROM person p1 USING person p2
+WHERE p1.id > p2.id
+    AND p1.name = p2.name
+    AND p1.instance_id = p2.instance_id;
```

Comment:
To allow `idx_person_local_instance` to be used:

```suggestion
DELETE FROM person p1 USING person p2
WHERE p1.id > p2.id
    AND p1.name = p2.name
    AND p1.instance_id = p2.instance_id AND NOT (p1.local OR p2.local);
```